aip/general/0162.md

```diff
@@ -25,201 +25,170 @@ to the version of an API as a whole.
 
 ## Guidance
 
-APIs **may** store a revision history for a resource if it is useful to users.
+APIs **may** store a revision history for a resource. Examples of when it is
+useful include:
 
-APIs implementing resources with a revision history **must** provide a
-`revision_id` field on the resource:
+- When it is valuable to expose older versions of a resource via an API. This
+  can avoid the overhead of the customers having to write their own API to store
+  and enable retrieval of revisions.
+- Other resources depend on different revisions of a resource.
+- There is a need to visualize the change of a resource over time.
 
-```proto
-message Book {
-  // The name of the book.
-  string name = 1;
+APIs implementing resources with a revision history **should** abstract resource
+revisions as nested collection of the resource. Sometimes, the revisions
+collection can be a top level collection, exceptions include:
+
+- If resource revisions are meant to have longer lifespan than the parent
+resource. In other words, resource revisions exist after the resource deletion.
 
-  // Other fields…
+The resource revision type **must** inherit from the parent resource and has a
```

Comment:
We've still got "inherit" here, which I don't think is helpful without elaboration.

Reply:
strange, I thought I updated this.

I'll merge and ensure it's fixed before resolving again.